### PR TITLE
Issue767 new pv system model

### PR DESCRIPTION
**Pull request by LauraMaier**
This is the PR for issue767.
A new pv system model was introduced which was validated using NIST measurement data.

The model offers a numerical and an analytical approach to estimate PV power generation and incorporates two different cell temperature calculation methods in order to take into account different mountings.

### CODE REVIEW

**Comment by LauraMaier** [comment]
Things To Do before merging:

- [ ] Check if implemented model AixLib.BoundaryConditions.SolarGeometry.BaseClasses.DeclinationSpencer can be exchanged by existing model AixLib.BoundaryConditions.SolarGeometry.BaseClasses.Declination
- [ ] Check if model AixLib.BoundaryConditions.SolarGeometry.BaseClasses.IncidenceAngleDuffie can be exchanged by existing model AixLib.BoundaryConditions.SolarGeometry.BaseClasses.IncidenceAngle
- Transfer existing records to new format (if available):
- [ ] AleoS24
- [ ] CanadianSolarCS6P250P
- [ ] SchuecoSME1
- [ ] SymphonyEnergySE6M181

File: AixLib/Electrical/PVSystem/BaseClasses/package.mo [hunk 1]
```diff
@@ -3,4 +3,140 @@ package BaseClasses
...
+  // Adjustable input parameters
+
+   parameter Real n_mod(final quantity=
+      "NumberOfModules", final unit="1") "Number of connected PV modules"
+      annotation ();
+
+
+  // Parameters from module data sheet
+
+   final parameter Modelica.SIunits.Efficiency eta_0=data.eta_0
```
Comment (Dawac) [inline AixLib/Electrical/PVSystem/BaseClasses/package.mo:23]: How well can all of these parameters be extracted from datasheets?
Maybe the parameters in the record could be clustered to geometrical, electrical and general parameters for easier readability.

**Review by Dawac — CHANGES_REQUESTED** [review]
Comment: I would add Revision data to the models and records, that were modified within the process of remodelling the PV-models.

Changes: I commented for some models, that somehow only some new lines were randomly added, maybe these files don't need to be changed. One result file with unrelated results seems to have been deleated, please double check if this was on purpose.

General Questions:
- How were the existing records converted?
- Is the assumption of Tau*Alpha = 0,9 always reasonable, since its not adjustable. 
- Would it make sense to implement a "dirt-factor" or something like that?

File: AixLib/Fluid/Actuators/package.mo [hunk 2]
```diff
@@ -32,14 +32,50 @@ package UsersGuide "User's Guide"
 
 
 
+
```
[the remark this answers is not in the document]
Comment (LauraMaier) [reply AixLib/Fluid/Actuators/package.mo:35]: Merged development and these changes are ignored now ;-)

File: AixLib/Media/AirIncompressible.mo [hunk 3]
```diff
@@ -391,7 +391,7 @@ algorithm
 "<html><p>
   This function computes the derivative of density with respect to
   temperature at constant pressure.
-</p>
+</p>/html /html
```
[the remark this answers is not in the document]
Comment (LauraMaier) [reply AixLib/Media/AirIncompressible.mo:394]: Same here. Solved by merging development

File: AixLib/Media/Specialized/Water/ConstantProperties_pT/package.mo [hunk 4]
```diff
@@ -1,133 +1,145 @@
...
-          extent={{20,-30},{78,-88}},
-          lineColor={0,0,0},
-          fillPattern=FillPattern.Sphere,
-          fillColor={95,95,95}),
-        Text(
-          extent={{-110,40},{50,-36}},
-          lineColor={238,46,47},
-          textString="pT")}));
-end ConstantProperties_pT;
+within AixLib.Media.Specialized.Water;
```
[the remark this answers is not in the document]
Comment (LauraMaier) [reply AixLib/Media/Specialized/Water/ConstantProperties_pT/package.mo:1]: Solved by merge of development

File: AixLib/DataBase/SolarElectric/AleoS24185.mo [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+<h4>
+  <span style=\"color: #008000\">References</span>
+</h4>
+<p>
+  Record for record used with <a href=
+  \"modelica://AixLib.Fluid.Solar.Electric.PVsystem\">AixLib.Fluid.Solar.Electric.PVsystem</a>
+</p>
+</html>",
+      revisions="<html><ul>
+  <li>
```
[the remark this answers is not in the document]
Comment (LauraMaier) [reply AixLib/DataBase/SolarElectric/AleoS24185.mo:34]: Done

File: AixLib/Types/Tilt/package.mo [hunk 6]
```diff
@@ -1,36 +1,45 @@
-within AixLib.Types;
```
[the remark this answers is not in the document]
Comment (LauraMaier) [reply AixLib/Types/Tilt/package.mo:1]: see resolved after merge

File: AixLib/Resources/ReferenceResults/Dymola/AixLib_AirFlow_AirHandlingUnit_Examples_AHU.txt [hunk 7]
```diff
@@ -1,28 +0,0 @@
-last-generated=2019-03-27
```
[the remark this answers is not in the document]
Comment (LauraMaier) [reply AixLib/Resources/ReferenceResults/Dymola/AixLib_AirFlow_AirHandlingUnit_Examples_AHU.txt:1]: Solved by merge of development

File: AixLib/Electrical/PVSystem/BaseClasses/package.mo (hunk 1 above)
Comment (LauraMaier) [reply AixLib/Electrical/PVSystem/BaseClasses/package.mo:23]: Done. The parameters are common manufacturer data. If data is not available default values were added for better usability :-)

**Comment by LauraMaier** [comment]
> Comment: I would add Revision data to the models and records, that were modified within the process of remodelling the PV-models.
> 
--> Changes done: added revision to all the models
> Changes: I commented for some models, that somehow only some new lines were randomly added, maybe these files don't need to be changed. One result file with unrelated results seems to have been deleated, please double check if this was on purpose.
> 
--> As discussed, this was partially caused by the html checker. To some parts, I revised it.

> General Questions:
> 
> * How were the existing records converted?
--> We converted them based on manufacturer data available online.

> * Is the assumption of Tau*Alpha = 0,9 always reasonable, since its not adjustable.
--> I introduced it as a final parameter for the first stage as it is not manufacturer data parameter, which users will know. It applies to Poly-Si-models with glass glazing.

> * Would it make sense to implement a "dirt-factor" or something like that?
--> As discussed, it would make sense in the future. Yet, this parameter really depends on the plant of interest and is determined rather random. Therefore, as a first step, I would like to not include it in the base model.

**Review by Dawac — APPROVED** [review]
Thanks for the effort! All issues have been adressed.

**Pull request merged**